frontend/controller/app/identity.js

```diff
@@ -0,0 +1,252 @@
+'use strict'
+
+import { GIErrorUIRuntimeError, L, LError, LTags } from '@common/common.js'
+import sbp from '@sbp/sbp'
+import { LOGIN, LOGIN_ERROR } from '~/frontend/utils/events.js'
+import { Secret } from '~/shared/domains/chelonia/Secret.js'
+import { boxKeyPair, buildRegisterSaltRequest, computeCAndHc, decryptContractSalt, hash, hashPassword, randomNonce } from '~/shared/zkpp.js'
+// Using relative path to crypto.js instead of ~-path to workaround some esbuild bug
+import { CURVE25519XSALSA20POLY1305, EDWARDS25519SHA512BATCH, deriveKeyFromPassword, serializeKey } from '../../../shared/domains/chelonia/crypto.js'
+import { handleFetchResult } from '../utils/misc.js'
+import { cloneDeep } from '@model/contracts/shared/giLodash.js'
+import { LOGIN_COMPLETE } from '../../utils/events.js'
+
+const loadState = async (identityContractID: string, password: ?string) => {
+  if (password) {
+    const stateKeyEncryptionKeyFn = (stateEncryptionKeyId, salt) => {
+      return deriveKeyFromPassword(CURVE25519XSALSA20POLY1305, password, salt + stateEncryptionKeyId)
+    }
+
+    const { encryptionParams, value: state } = await sbp('gi.db/settings/loadEncrypted', identityContractID, stateKeyEncryptionKeyFn)
+
+    if (state) {
+      const cheloniaState = state.cheloniaState
+      delete state.cheloniaState
```

Comment:
Where in this project is it clearly explained/defined what is inside of `cheloniaState`?

Reply:
It wasn't, but I've now added comments explaining it.